diff --git a/boringssl/BUILD.gn b/boringssl/BUILD.gn
index 5691d26cfc2..b590c092743 100644
--- a/boringssl/BUILD.gn
+++ b/boringssl/BUILD.gn
@@ -109,7 +109,7 @@ if (enable_rust_boringssl) {
   }
 }
 
-if (is_msan) {
+if (is_msan || current_cpu == "arm64") {
   # MSan instrumentation is incompatible with assembly optimizations.
   # BoringSSL's GAS-compatible assembly knows how to detect MSan, but the NASM
   # assembly does not, so we check for MSan explicitly.
diff --git a/crc32c/BUILD.gn b/crc32c/BUILD.gn
index d2a79338adf..d76633384e4 100644
--- a/crc32c/BUILD.gn
+++ b/crc32c/BUILD.gn
@@ -39,7 +39,7 @@ config("crc32c_config") {
     defines += [ "HAVE_BUILTIN_PREFETCH=0" ]
   }
 
-  if (current_cpu == "arm64") {
+  if (current_cpu == "arm64" && !is_win) {
     defines += [ "HAVE_ARM64_CRC32C=1" ]
   } else {
     defines += [ "HAVE_ARM64_CRC32C=0" ]
diff --git a/libaom/BUILD.gn b/libaom/BUILD.gn
index 187228acbe1..9c61f0114bc 100644
--- a/libaom/BUILD.gn
+++ b/libaom/BUILD.gn
@@ -54,6 +54,8 @@ if (is_nacl) {
   if (is_posix || is_fuchsia) {
     # Should cover linux, fuchsia, mac, and ios.
     os_category = "linux"
+  } else if (is_win && current_os == "winuwp") {
+    os_category = "win"
   } else {  # This should only match windows.
     os_category = current_os
   }
diff --git a/libvpx/BUILD.gn b/libvpx/BUILD.gn
index 7c2b3c3e0dc..11f71cde0fe 100644
--- a/libvpx/BUILD.gn
+++ b/libvpx/BUILD.gn
@@ -59,6 +59,8 @@ if (is_nacl) {
   } else if (is_posix || is_fuchsia) {
     # Should cover linux, fuchsia, mac, and the ios simulator.
     os_category = "linux"
+  } else if (current_os == "winuwp") {
+    os_category = "win"
   } else {  # This should only match windows.
     os_category = current_os
   }
diff --git a/pffft/BUILD.gn b/pffft/BUILD.gn
index 0203735ac90..ce74d354e93 100644
--- a/pffft/BUILD.gn
+++ b/pffft/BUILD.gn
@@ -7,7 +7,9 @@ import("//testing/libfuzzer/fuzzer_test.gni")
 import("//testing/test.gni")
 
 config("common_config") {
-  cflags = [ "-Wno-shadow" ]
+  if (is_clang) {
+    cflags = [ "-Wno-shadow" ]
+  }
 
   if (is_win) {
     defines = [
