diff --git a/boringssl/BUILD.gn b/boringssl/BUILD.gn
index efcbdf378b6..8809b1d7657 100644
--- a/boringssl/BUILD.gn
+++ b/boringssl/BUILD.gn
@@ -65,7 +65,7 @@ if (is_win && !is_msan && current_cpu != "arm64") {
     asmflags = []
     include_dirs = [ "src/include" ]
 
-    if (is_msan) {
+    if (is_msan || current_cpu == "arm64") {
       public_configs = [ ":no_asm_config" ]
     } else if (current_cpu == "x64") {
       if (is_apple) {
diff --git a/crc32c/BUILD.gn b/crc32c/BUILD.gn
index 38024bbd8d0..e3281474c72 100644
--- a/crc32c/BUILD.gn
+++ b/crc32c/BUILD.gn
@@ -40,7 +40,7 @@ config("crc32c_config") {
     defines += [ "HAVE_BUILTIN_PREFETCH=0" ]
   }
 
-  if (current_cpu == "arm64") {
+  if (current_cpu == "arm64" && !is_win) {
     defines += [ "HAVE_ARM64_CRC32C=1" ]
   } else {
     defines += [ "HAVE_ARM64_CRC32C=0" ]
diff --git a/libvpx/BUILD.gn b/libvpx/BUILD.gn
index 20be02b87d9..c26c38cbe7f 100644
--- a/libvpx/BUILD.gn
+++ b/libvpx/BUILD.gn
@@ -53,6 +53,8 @@ if (is_nacl) {
   } else if (is_posix || is_fuchsia) {
     # Should cover linux, fuchsia, mac, and the ios simulator.
     os_category = "linux"
+  } else if (current_os == "winuwp") {
+    os_category = "win"
   } else {  # This should only match windows.
     os_category = current_os
   }
diff --git a/pffft/BUILD.gn b/pffft/BUILD.gn
index 322c00dbd2d..84610f3523c 100644
--- a/pffft/BUILD.gn
+++ b/pffft/BUILD.gn
@@ -7,7 +7,9 @@ import("//testing/libfuzzer/fuzzer_test.gni")
 import("//testing/test.gni")
 
 config("common_config") {
-  cflags = [ "-Wno-shadow" ]
+  if (is_clang) {
+    cflags = [ "-Wno-shadow" ]
+  }
 
   if (is_win) {
     defines = [
