From 674aa02ce6cfb948f3e68f181bce939e9fab8336 Mon Sep 17 00:00:00 2001
From: Fela <frayx@live.it>
Date: Wed, 1 Oct 2025 20:45:27 +0400
Subject: [PATCH 1/4] Fix build UWP

---
 extras/package/win32/package.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/package/win32/package.mak b/extras/package/win32/package.mak
index bd30ad39eb..443c771324 100644
--- a/extras/package/win32/package.mak
+++ b/extras/package/win32/package.mak
@@ -94,7 +94,7 @@ package-win-npapi: build-npapi
 	mkdir -p "$(win32_destdir)/sdk/activex/"
 	cp $(top_builddir)/npapi-vlc/activex/README.TXT $(top_builddir)/npapi-vlc/share/test/test.html $(win32_destdir)/sdk/activex/
 
-package-win-strip: package-win-common package-win-npapi
+package-win-strip: package-win-common
 	mkdir -p "$(win32_debugdir)"/
 	find $(win32_destdir) -type f \( -name '*$(LIBEXT)' -or -name '*$(EXEEXT)' \) | while read i; \
 	do if test -n "$$i" ; then \
-- 
2.40.0.windows.1

