From 3becf6173b42d4e23773f56b847c622a82d8fa34 Mon Sep 17 00:00:00 2001
From: Fela <frayx@live.it>
Date: Wed, 1 Oct 2025 20:45:49 +0400
Subject: [PATCH 2/4] Print plugin file name on load

---
 src/modules/modules.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/modules/modules.c b/src/modules/modules.c
index 7d29370e4a..7793180369 100644
--- a/src/modules/modules.c
+++ b/src/modules/modules.c
@@ -318,8 +318,8 @@ done:
 
     if (module != NULL)
     {
-        msg_Dbg (obj, "using %s module \"%s\"", capability,
-                 module_get_object (module));
+        msg_Dbg (obj, "using %s module \"%s\" from %s", capability,
+                 module_get_object (module), module->plugin->path);
         vlc_object_set_name (obj, module_get_object (module));
     }
     else
-- 
2.40.0.windows.1

