From 7c5ae01b98180bfc60195857146d34f2bade7a89 Mon Sep 17 00:00:00 2001
From: Fela <frayx@live.it>
Date: Tue, 7 Oct 2025 18:16:24 +0400
Subject: [PATCH 4/4] Enable fast seek for imem access

---
 modules/access/imem-access.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/modules/access/imem-access.c b/modules/access/imem-access.c
index 25547c1490..355a529c6a 100644
--- a/modules/access/imem-access.c
+++ b/modules/access/imem-access.c
@@ -75,11 +75,8 @@ static int Control(stream_t *access, int query, va_list args)
     switch (query)
     {
         case STREAM_CAN_SEEK:
-            *va_arg(args, bool *) = sys->seek_cb != NULL;
-            break;
-
         case STREAM_CAN_FASTSEEK:
-            *va_arg(args, bool *) = false;
+            *va_arg(args, bool *) = sys->seek_cb != NULL;
             break;
 
         case STREAM_CAN_PAUSE:
-- 
2.40.0.windows.1

